parser/scrape.py

```diff
@@ -0,0 +1,83 @@
+# Testing beautiful soup for scraping PDFs from FE website
+
+import requests 
+from bs4 import BeautifulSoup
+import re
+import os
+import json
+
+json_output = {"documents":{}}
+
+url = "https://www.cs.ucf.edu/registration/exm/"
+read = requests.get(url) # Get request for HTML
+html = read.content
+soup = BeautifulSoup(html, "html.parser")
+
+# Is the tag a PDF link for an exam that has no discrete section?
+def pdf_before_discrete(tag):
+    return tag.name == "a" and re.search(".pdf", tag["href"]) and not re.search("CS|DS", tag["href"])
+
+# Return a string containing the semester and year of the exam specified
+def get_semester_and_year(file_name):
```

Comment:
**Logic Error:**
The `get_semester_and_year()` function logic is fundamentally flawed since you are classifying the FE semester based solely on the file name from the website path. Examine the dates on the post-discrete FEs we are using. Spring exams have been held in January and May while Summer exams have been held as late as August, not accounting for covid edge cases. 

**Suggestions:**
Use Python's datetime library, the weeks in a semester (considering Spring/Fall and Summer differences), and account for the weeks off in between each semester, to then track with an accuracy of, at least to the month, when the FEs have been, and will be held in the future. 

Let me know if you have any questions!

Reply:
Any semester distinction is only being used for unique key labeling purposes and nothing more right now. I'll modify labeling conventions for the time being to make that more clear.

As it stands right now, the completely accurate semester designation seems to only be available in the actual PDF contents themselves, which the scraper does not currently access. 

I can tighten the accuracy of this approach, but if we need 100% accuracy, I don't think the scraper is the best place to make any real semester distinctions. There could always be unpredictable edge cases that crop up in the future in regards to dates of exams not falling when we expect.

Do we think semester extraction could/should be added to the parser code? If so, I'll gladly add it! I'm also open to discuss other ideas.

Reply:
Found another, similar issue. I need to rework the logic that distinguishes which exams are pre/post discrete era. Unlike the semester distinction, I believe this will be doable in the scraper since the cutoff date when the discrete section was removed is set in stone.

Reply:
Found a way to extract semester in a part of the HTML that I overlooked. Will add this but I think we should still verify.